pkg/controller/job/job_controller_test.go

```diff
@@ -1894,10 +1894,6 @@ func TestPastDeadlineJobFinished(t *testing.T) {
 				t.Errorf("Could not create Job: %v", err)
 			}
 
-			if err := sharedInformerFactory.Batch().V1().Jobs().Informer().GetIndexer().Add(job); err != nil {
```

Comment:
:thinking: the unit test didn't used to work without these calls to the Indexer. Maybe the test client changed?

Reply:
IIRC these lines were added in an attempt to get rid of flakes before we understood the underlying cause for the flakes: https://github.com/kubernetes/kubernetes/pull/114944#discussion_r1069179801. 

To check that, I have checked out the initial commits ([commit1](https://github.com/kubernetes/kubernetes/commit/4c1b95ddfafee38a8acd654aaec6e00829b3908b) and [commit2](https://github.com/kubernetes/kubernetes/commit/eea7dca085af3b32199b2d4210d229c2dd624791)) introducing these lines, removed it and the tests passed (but wouldn't pass consistently).